pkg/vul/import.go

```diff
@@ -60,14 +65,34 @@ func Import(ctx context.Context, opt *types.VulnerabilityOptions) error {
 		return errors.New("expected non-nil result")
 	}
 
-	for noteID, nocc := range list {
-		log.Debug().Msgf("note: %s, occurrences: %d", noteID, len(nocc.Occurrences))
-		if err := postNoteOccurrences(ctx, opt.Project, noteID, nocc); err != nil {
-			return errors.Wrap(err, "error posting notes")
+	resultCh := make(chan string, occurrencePostChannelSize)
+	exitCh := make(chan error)
+
+	var wg sync.WaitGroup
+
+	go func() {
```

Comment:
I don't think this nested goroutine is necessary here. It looks like you are trying to use the outer goroutine to exit if there is a single error. A better way to handle it would be to use contexts.

Something like

ctx, cancel := context.WithCancel(ctx)
defer ceancel()

for note... {
  wg.Add()
  go func(...) {
    defer wg.Done()
    select {
    case <-ctx.Done():
    default:
    }
    if err := postNoteOccurrences(...); err != nil {
      cancel()
      return
    }
  }(...)
}

Reply:
Yep, cleaner and same perf.